backend/src/ee/routes/v1/identity-project-additional-privilege-router.ts

```diff
@@ -0,0 +1,265 @@
+import { MongoAbility, RawRuleOf } from "@casl/ability";
+import { PackRule, packRules, unpackRules } from "@casl/ability/extra";
+import slugify from "@sindresorhus/slugify";
+import ms from "ms";
+import { z } from "zod";
+
+import { IdentityProjectAdditionalPrivilegeSchema } from "@app/db/schemas";
+import { IdentityProjectAdditionalPrivilegeTemporaryMode } from "@app/ee/services/identity-project-additional-privilege/identity-project-additional-privilege-types";
+import { ProjectPermissionSet } from "@app/ee/services/permission/project-permission";
+import { IDENTITY_ADDITIONAL_PRIVILEGE } from "@app/lib/api-docs";
+import { alphaNumericNanoId } from "@app/lib/nanoid";
+import { verifyAuth } from "@app/server/plugins/auth/verify-auth";
+import { AuthMode } from "@app/services/auth/auth-type";
+
+export const registerIdentityProjectAdditionalPrivilegeRouter = async (server: FastifyZodProvider) => {
+  server.route({
+    url: "/",
+    method: "POST",
+    schema: {
+      body: z.union([
+        z.object({
+          identityId: z.string().min(1).describe(IDENTITY_ADDITIONAL_PRIVILEGE.CREATE.identityId),
+          projectSlug: z.string().min(1).describe(IDENTITY_ADDITIONAL_PRIVILEGE.CREATE.projectSlug),
+          slug: z
+            .string()
+            .min(1)
+            .max(60)
+            .trim()
+            .default(`privilege-${slugify(alphaNumericNanoId(12))}`)
+            .refine((val) => val.toLowerCase() === val, "Must be lowercase")
+            .refine((v) => slugify(v) === v, {
+              message: "Slug must be a valid slug"
+            })
+            .describe(IDENTITY_ADDITIONAL_PRIVILEGE.CREATE.slug),
+          permissions: z.any().array().describe(IDENTITY_ADDITIONAL_PRIVILEGE.CREATE.permissions),
+          isPackedPermission: z
```

Comment:
What is this used for? Can we just allow un-packed for all so API users don't have to decide? I think having to choose which type will create confusion for api users

Reply:
By default users don't have to think for that's why its false so that if they don't send it. They don't have to care about. 

Packed is casl packing permissions. It reduces the permission json object size by 50%. In server we save it as packed. That's why we set the option. In dashboard we already pack it from our side so server don't need to pack it